### PR TITLE
Change routeguide example to use proto 3.0.0

### PR DESCRIPTION
**Pull request by zsurocking**
Also add back the deleted .gitignore to helloworld example.

### CODE REVIEW

**Comment by carl-mastrangelo** [comment]
@zsurocking LGTM

File: examples/android/helloworld/app/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (ejona86) [inline examples/android/helloworld/app/.gitignore]: How did you notice this? This shouldn't actually do anything because "build" is ignored at the root.

**Comment by ejona86** [comment]
The `build.gradle` change LGTM.

File: examples/android/helloworld/app/.gitignore (hunk 1 above)
Comment (zsurocking) [inline examples/android/helloworld/app/.gitignore]: Ha, you're right. Removed.

**Comment by ejona86** [comment]
Backported in #2150

**Pull request merged**